src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestNode.cs

```diff
@@ -70,10 +74,9 @@ internal class HttpRequestNode : HttpSyntaxNode
     public HttpBindingResult<HttpRequestMessage> TryGetHttpRequestMessage(HttpBindingDelegate bind)
     {
         var request = new HttpRequestMessage();
-        var diagnostics = new List<Diagnostic>();
-        var success = true;
+        var diagnostics = new List<Diagnostic>(base.GetDiagnostics());
```

Comment:
The binding methods for the child nodes (such as TryGetUri and TryGetBody) should also call base.GetDiagnostics() to include syntax errors (and return false if there are any diagnostics with severity Error).

Reply:
I guess that could lead to some duplication at higher levels (unless we change the list to a hashset)...

Reply:
Yeah, I deliberately kept these concerns separate.

Reply:
Thinking more perhaps it may be better for the binding API calls to only return binding errors. The syntax errors are already available on the node which the caller is trying to bind.

This may be more flexible from an API standpoint and would also keep the behavior consistent regardless of which construct is being bound... Otherwise, it forces callers to reason about which construct they are binding and whether or not they should fetch the syntax diagnostics separately.